src/main/java/uk/gov/hmcts/cmc/claimstore/services/InterestRateService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.exceptions.BadRequestException;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+
+@Component
+public class InterestRateService {
+
+    public BigDecimal calculateRate(LocalDate fromDate, LocalDate toDate, BigDecimal rate, BigDecimal amount) {
+        validateVarIsNonNegativeBigDecimal(rate);
+        validateVarIsNonNegativeBigDecimal(amount);
+
+        long noOfDays = calculateNumberOfDays(fromDate, toDate);
+
+        return amount
+            .multiply(rate)
+            .multiply(BigDecimal.valueOf(noOfDays))
+            .divide(BigDecimal.valueOf(365 * 100), 2, RoundingMode.HALF_UP);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS2184)